src/main/java/io/stargate/sgv2/dynamoapi/dynamo/ItemProxy.java

```diff
@@ -134,4 +141,98 @@ public PutItemResult putItem(PutItemRequest putItemRequest, StargateBridgeClient
+      assert clusteringKeys.size() == 1;
+      clusteringKey = Optional.of(clusteringKeys.get(0));
+    }
+
+    // Query first
+    QueryBuilder.QueryBuilder__21 queryBuilder =
+        new QueryBuilder()
+            .select()
+            .from(KEYSPACE_NAME, tableName)
+            .where(
+                partitionKey.getName(),
+                Predicate.EQ,
+                DataMapper.fromDynamo(keyMap.get(partitionKey.getName())));
+    if (clusteringKey.isPresent()) {
+      final String clusterKeyName = clusteringKey.get().getName();
+      queryBuilder =
+          queryBuilder.where(
+              clusterKeyName, Predicate.EQ, DataMapper.fromDynamo(keyMap.get(clusterKeyName)));
+    }
+
+    QueryOuterClass.Response response = bridge.executeQuery(queryBuilder.build());
+    Collection<Map<String, AttributeValue>> resultRows =
+        collectResults(response.getResultSet(), "", new HashMap<String, String>());
+    // if no result, exit
+    if (resultRows.isEmpty()) {
+      return new DeleteItemResult();
+    }
+    // DeleteItem only affect 1 row. Multiple delete need other requests combination (query+delete /
+    // BatchWrite)
+    assert resultRows.size() == 1;
+
+    // Regard ConditionExpression as FilterExpression
+    // Construct Predicate
+    java.util.function.Predicate<Map<String, AttributeValue>> predicate;
+    if (conditionExpression.isEmpty()) {
+      predicate = item -> true;
+    } else {
+      CharStream chars = CharStreams.fromString(conditionExpression);
+      Lexer lexer = new ExpressionLexer(chars);
+      CommonTokenStream tokens = new CommonTokenStream(lexer);
+      ExpressionParser parser = new ExpressionParser(tokens);
+      FilterExpressionVisitor visitor =
+          new FilterExpressionVisitor(
+              deleteItemRequest.getExpressionAttributeNames(),
+              deleteItemRequest.getExpressionAttributeValues());
+      ParseTree tree = parser.expr();
+      predicate = item -> visitor.isMatch(tree, item);
+    }
+    List<Map<String, AttributeValue>> resultList =
+        resultRows.stream().filter(predicate).collect(Collectors.toList());
+
+    // if no result, exit
+    if (resultList.isEmpty()) {
+      return new DeleteItemResult();
+    }
+
+    QueryBuilder.QueryBuilder__43 deleteBuilder =
+        new QueryBuilder()
+            .delete()
+            .from(KEYSPACE_NAME, tableName)
+            .where(
+                partitionKey.getName(),
+                Predicate.EQ,
+                DataMapper.fromDynamo(keyMap.get(partitionKey.getName())));
+
+    if (clusteringKey.isPresent()) {
+      final String clusterKeyName = clusteringKey.get().getName();
+      deleteBuilder =
+          deleteBuilder.where(
+              clusterKeyName, Predicate.EQ, DataMapper.fromDynamo(keyMap.get(clusterKeyName)));
+    }
+    bridge.executeQuery(deleteBuilder.build());
+    final String returnValues = deleteItemRequest.getReturnValues();
+    if (returnValues.equals("ALL_OLD")) {
```

Comment:
Please use `com.amazonaws.services.dynamodbv2.model.ReturnValue.ALL_OLD`.

Reply:
Got it.